remotefs/download.go

```diff
@@ -0,0 +1,51 @@
+package remotefs
+
+import (
+	"crypto/sha256"
+	"encoding/hex"
+	"fmt"
+	"io"
+	"os"
+)
+
+// ErrChecksumMismatch is returned when the checksum of the uploaded file does not match the local checksum.
+
+// Download a file from the remote host.
+func Download(fs FS, src, dst string) error {
+	remote, err := fs.Open(src)
+	if err != nil {
+		return fmt.Errorf("open remote file for download: %w", err)
+	}
+	defer remote.Close()
+
+	remoteStat, err := remote.Stat()
+	if err != nil {
+		return fmt.Errorf("stat remote file for download: %w", err)
+	}
+
+	remoteSum := sha256.New()
+	localSum := sha256.New()
+
+	local, err := os.OpenFile(dst, os.O_CREATE|os.O_WRONLY|os.O_TRUNC, remoteStat.Mode())
```

Comment:
This could check if the file exists locally and compare the shasums and just return nil if they match already 🤔

Reply:
(and same for upload)

Reply:
Maybe for another pr

Reply:
I was wondering that too, but we write the sums on copy, we'd have to split it up(to first read the sums compare and then download/upload if we need to, which could take longer than just uploading/downloading(I assume most uploaded/downloaded files won't be larger than a few megs.

```golang
if _, err := io.Copy(io.MultiWriter(local, localSum), remoteReader); err != nil {
		_ = local.Close()
		return fmt.Errorf("copy file from remote host: %w", err)
```

Reply:
In k0sctl the files can be hundreds of megabytes. Currently k0sctl checks for file size + modification time to avoid re-uploading.